go/libraries/doltcore/sqle/statspro/doc.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2025 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statspro
+
+// Package statspro provides a queue that manages table statistics
+// management and access.
+//
+// At any given time there is one work generating thread, one scheduling
+// thread, and one execution thread.
+//
+// The worker loops fetching the most recent session root,
+// reading all of its databases/tables/ indexes, collecting statistics
+// for those objects, and updating the shared statistics state. Every
+// cycle replaces the shared state.
+//
+// Work is delegated to the scheduler thread, which serializes
+// issuer jobs with concurrent async requests, and rate limits sending
+// jobs to the execution thread. The execution thread completes
+// function callbacks.
+//
+// GC occurs within an update cycle. Through a cycle GC populates an
+// in-memory cache with the complete and exclusive set of values of
+// the new shared statistics object. Both are atomically swapped using
+// a generation counter (which may or may not be necessary, but is one
+// of several guards against surprising concurrent changes).
+//
+// Concurrent issuer threads are further restrained with a context list
+// that at most one thread owns. There are two contexts, one for the
+// thread and another for the specific update cycle. Listeners (like wait)
+// use the second context to follow update cycles. Concurrent restarts
+// cancel and replace the previous owner's contexts with their own. Atomic
+// shared state swaps are likewise guarded on the issuer's context
+// integrity.
+//
+// All stats are persisted within a single database. If there are
+// multiple databases, one is selected by random as the storage target.
```

Comment:
What does this mean?

Reply:
This sounds like we're storing stats inline with other ref data on one of the DBs being analyzed. Doesn't that break root hash equivalence checks for people who care about that?